src/Symfony/EventListener/ReadListener.php

```diff
@@ -60,6 +65,21 @@ public function onKernelRequest(RequestEvent $event): void
         $request = $event->getRequest();
         $operation = $this->initializeOperation($request);
 
+        /** @var \Exception $exception */
+        if (($exception = $request->attributes->get('exception')) && $operation?->getExtraProperties()['hydra_errors']) {
+            $operation = $this->initializeOperationForException($exception);
+            if (null === $operation) {
+                $format = ErrorFormatGuesser::guessErrorFormat($request, [
+                    'jsonproblem' => ['mime_types' => ['application/problem+json']],
+                    'jsonld' => ['mime_types' => ['application/ld+json']],
+                ]);
+                $operation = match ($format['key']) {
+                    'jsonproblem' => $this->initializeOperationForException(ProblemError::createFromException($exception, $exception->getCode())),
+                    default => $this->initializeOperationForException(Error::createFromException($exception, $exception->getCode())),
+                };
```

Comment:
Format negotiation shouldn't be here. Specify the format of each error resource (ProblemError and Error) and let the router do this.